cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -709,11 +720,76 @@ fn ty_from_ty_hint_or_reg_class<M: ABIMachineSpec>(r: Reg, ty: Option<Type>) ->
     }
 }
 
+fn gen_move_multi<M: ABIMachineSpec>(
+    dst: ValueRegs<Writable<Reg>>,
+    src: ValueRegs<Reg>,
+    ty: Type,
+) -> Vec<M::I> {
```

Comment:
/me wears the premature optimization hat

Could we use `SmallVec<[M::I; VALUE_REGS_PARTS]>` here, and for the other helpers (and some of their callers too)? It documents the 1:1 mapping between move/load/store Inst and real reg present in the value-regs, and may even reduce the number of dynamic allocations.

Reply:
Done, sort of; this type is exposed up to the ABI trait and I'm not sure I want to expose a detail like "max number of value registers per value" there, but it does seem reasonable to pick a small-ish SmallVec size and standardize it; so I went through the ABI code with a `SmallVec<[Inst; 4]>`-shaped paintbrush using a new `SmallInstVec<Inst>` typedef.